cpp/inc/bond/ext/grpc/thread_pool.h

```diff
@@ -62,7 +44,7 @@ class thread_pool : private boost::asio::io_service
     /// will be created.
     explicit
     thread_pool(size_t numThreads = USE_HARDWARE_CONC)
-        : _work(*this)
+        : _service{ std::make_shared<service>() }
```

Comment:
Let's split this into two ctor and get rid of the `USE_HARDWARE_CONC` constant completely.